akka-testkit-typed/src/main/scala/akka/testkit/typed/javadsl/TestKitJunitResource.scala

```diff
@@ -95,4 +99,31 @@ class TestKitJunitResource(_kit: ActorTestKit) extends ExternalResource {
     testKit.shutdownTestKit()
   }
 
+  /**
+   * Execute code block while bounding its execution time between `min` and
+   * `max`. `within` blocks may be nested. All methods in this trait which
+   * take maximum wait times are available in a version which implicitly uses
+   * the remaining time governed by the innermost enclosing `within` block.
+   *
+   * Note that the timeout is scaled using Duration.dilated, which uses the
+   * configuration entry "akka.actor.typed.test.timefactor", while the min Duration is not.
+   *
+   * {{{
+   * val ret = within(50 millis) {
+   *   test ! Ping
+   *   expectMessageType[Pong]
+   * }
+   * }}}
+   */
+  def within[T](min: FiniteDuration, max: FiniteDuration, f: Supplier[T]): T =
+    testKit.within(min, max, f)
+
+  /**
+   * Same as calling `within(0 seconds, max)(f)`.
+   */
+  def within[T](max: FiniteDuration, f: Supplier[T]): T =
+    testKit.within(Duration.Zero, max, f)
+
+  def setFinished(isFinished: Boolean): Unit = testKit.setFinished(isFinished)
+
```

Comment:
remove, same reason as for ActorTestKit

Reply:
Sorry for not coming back to this before.
Does this mean that the Within functionality should only exist in TestProbe?